parser/statementStartTokens.coffee

```diff
@@ -0,0 +1,17 @@
+class StartTokens
+  @statement = [
+    'for', 'while',                                  # Start of statement
+    @expression
```

Comment:
Yes, I did intend that. If you take a look at the way that `at` is implemented, it supports nested arrays.